### ISSUE TITLE
Narrator doesn't read Dialog's title and subText properly when changing dialog's host

### ISSUE THREAD

**Issue by eskomsft** [microsoft/fluentui#15378]
### Environment Information

- **Package version(s)**: 7.144.0
- **Browser and OS versions**: Edge 85.0.564.68 (Official build) (64-bit) Windows 19041.508

### Describe the issue:

To control where the Dialog is rendered, we use layerProps.hostId and position the Dialog using style: root: { top: 0, left: 0}
This makes the narrator read only the name of the first button on the Dialog.

### Please provide a reproduction of the issue in a codepen: https://codepen.io/eskomsft/pen/LYZPeJw

commenting out following line makes it work: layerProps: { hostId: 'dialogHostId' },

#### Actual behavior:

Narrator reads "Fabric example pen tab item - send button"

#### Expected behavior:

Narrator reads "Fabric example pen tab item - send button - Heading level 1 Missing subject - do you want to send this message without a subject "

### Documentation describing expected behavior

The examples at https://developer.microsoft.com/en-us/fluentui#/controls/web/dialog are all read as expected, so I expect that's how the Dialogs are supposed to be read.

**Comment by tomi-msft** [comment]
@eskomsft Thanks for the issue! We'll look into this more.

**Comment by paulgildea** [comment]
@smhigley FYI
